src/warning/warning-printer.ts

```diff
@@ -137,7 +137,7 @@ function getSquiggleUnderline(
     const endColumn = sourceRange.end.line === sourceRange.start.line ?
         sourceRange.end.column :
         lineText.length;
-    const prefix = ' '.repeat(startColumn);
+    const prefix = lineText.replace(/[^\t]/g, ' ').substr(0, startColumn);
```

Comment:
Oh, I see, yes! I misread the `[^\t]` part of your regex, easy to forget that `^` means to invert the character set. It should have been a clue that all the tests pass! :)

The only recommendations left would be to prefer the `slice` method to the `substr` method, and to do the slice before the replace, as a microoptimization.